### PR TITLE
WIT visual redesign part 3

### PR DESCRIPTION
**Pull request by tolga-b**
* Motivation for features / changes
Continuing visual redesign ahead of WIT 1.0 launch.

* Technical description of changes
Modified partial dependence plots section to look like mocks and match the style in performance tab.
Removed some unused functions
Reorganized callback functions for pd plots to accommodate changes in html
Miscellaneous style (color margin etc.) edits to other sections.

* Screenshots of UI changes
pdp

* Detailed steps to verify changes work correctly (as executed by you)
Ran tensorboard version of WIT with 1 and 2 binary classificationmodels and verified that all tabs are working correctly.
Need to do more detailed tests once all visual redesign PRs are in.
* Alternate designs / implementations considered

### CODE REVIEW

**Comment by tolga-b** [comment]
@jameswex please review.

File: tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-inference-viewer.html [hunk 1]
```diff
@@ -39,7 +39,9 @@
       .header {
         color: #80868B;
         font-size: 12px;
-        padding: 5px 0;
+        padding: 4px 0;
+        border-bottom: solid 1px #DADCE0;
+        border-top: solid 1px #DADCE0;
```
Comment (jameswex) [inline tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-inference-viewer.html]: is this from mahima feedback?

File: tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html [hunk 2]
```diff
@@ -84,17 +84,19 @@
 
       .example-holder {
         width: 100%;
+        margin-left: -5px;
+        margin-top: -5px;
```
Comment (jameswex) [inline tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html]: same question with these updateds

File: tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html [hunk 3]
```diff
@@ -304,14 +303,31 @@
...
+      .pd-range-hyphen {
+        padding-right: 6px;
+        padding-top: 8px;
+      }
+
+      .pd-range-control {
+        border-bottom: solid 1px #FCC934;
+        border-top: none;
+        border-left: none;
+        border-right: none;
```
Comment (jameswex) [inline tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html]: should have border-* none by default. are these necessary?

File: tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html [hunk 4]
```diff
@@ -375,53 +391,43 @@
...
         border-top: none;
         border-bottom-left-radius: 2px;
         border-bottom-right-radius: 2px;
         background: white;
       }
 
       :host vz-line-chart {
-        height: 120px;
-        width: 200px;
+        margin: 12px 12px 12px 24px;
```
Comment (jameswex) [inline tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html]: how does this affect roc curves?

File: tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html [hunk 5]
```diff
@@ -2404,6 +2428,8 @@ <h2>Show similarity to selected datapoint</h2>
           readOnly: true,
         },
         // Timer for debounding of refreshing dive/stats after data changes.
+        refreshPDPlotTimer_: Number,
+        // Timer for debounding of refreshing dive/stats after data changes.
```
Comment (jameswex) [inline tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html]: looks like i had a typo in my comment for the other timer. should be "debouncing"

File: tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-inference-viewer.html (hunk 1 above)
Comment (mpushkarna) [reply tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-inference-viewer.html]: Added a border around table headers in modules to distinguish them from any icons/buttons around them.

File: tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html (hunk 2 above)
Comment (mpushkarna) [reply tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html]: Added this for full-width feature value rows (was looking strange on small width arrangements)

File: tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html (hunk 3 above)
Comment (mpushkarna) [reply tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html]: This overrides the default borders on T/R/L and emphasizes the yellow border on the bottom. All other changes for color & grid consistency.

File: tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html (hunk 4 above)
Comment (mpushkarna) [reply tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html]: Adds consistent spacing around vz-line-chart for positioning.

File: tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html (hunk 5 above)
Comment (tolga-b) [reply tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html]: done.

File: tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html [hunk 6]
```diff
@@ -5189,38 +5205,70 @@ <h2>Show similarity to selected datapoint</h2>
...
         for (let i = 0; i < elements.length; i++) {
           this.hidePdElement_(elements[i]);
         }
       },
 
+      pdInputChanged: function(event) {
+        // Debounce a refresh request, for when multiple updates are happening
+        // in succession.
+        clearTimeout(this.refreshPDPlotTimer_);
+        this.refreshPDPlotTimer_ = setTimeout(() => this.pdInputChangedImpl(event), 500);
```
Comment (stephanwlee) [inline tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html]: prefer to use `this.debounce` https://polymer-library.polymer-project.org/1.0/api/classes/Polymer.Base#method-debounce

Comment (tolga-b) [reply tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html]: Done! Thank you for this, I wasn't aware of the native option.

File: tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html [hunk 7]
```diff
@@ -1005,32 +1020,35 @@
...
         width: 100%;
         display: flex;
-        border-bottom: 1px solid lightgrey;
-        border-top: 1px solid lightgrey;
-        border-radius: 5px;
-        box-shadow: 0 1px 4px #888888;
+        border-bottom: 1px solid #DADCE0;
+        border-top: 1px solid #DADCE0;
+        border-radius: 2px;
+        box-shadow: 0px 1px 2px 0px rgba(60,64,67,0.3), 0px 1px 3px 1px rgba(60,64,67,0.15);
```
[the remark this answers is not in the document]
Comment (tolga-b) [reply tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html]: done.

File: tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html [hunk 8]
```diff
@@ -84,17 +84,19 @@
 
       .example-holder {
         width: 100%;
+        margin-left: -5px;
```
[the remark this answers is not in the document]
Comment (tolga-b) [reply tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html]: removed it, thanks!

File: tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html [hunk 9]
```diff
@@ -84,17 +84,19 @@
...
-        margin-top: 5px;
-        margin-right: 5px;
+        margin-top: 6px;
+        margin-right: 6px;
       }
 
       .inference-section {
         width: 100%;
-        border: 1px solid lightgrey;
+        border: 1px solid #DADCE0;
```
[the remark this answers is not in the document]
Comment (tolga-b) [reply tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html]: done, changed to wit-color-gray300.

**Comment by stephanwlee** [comment]
Not sure if you have the write access but if you do, feel free to merge :)

**Pull request merged**